resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -687,6 +687,20 @@ private[spark] class Client(
   private def createConfArchive(): File = {
     val hadoopConfFiles = new HashMap[String, File]()
 
+    // SPARK_CONF_DIR shows up in the classpath before HADOOP_CONF_DIR/YARN_CONF_DIR
+    val localConfDir = System.getProperty("SPARK_CONF_DIR",
```

Comment:
`SPARK_CONF_DIR` is set by Spark's launch scripts, so you should just be able to do:

```
sys.env.get("SPARK_CONF_DIR").foreach { ... }
```

Reply:
not exactly till now  , plz check https://github.com/apache/spark/pull/19688

Reply:
So it's being fixed and you can apply my suggestion, no?

Reply:
ok, thanks for you advise